addons/pos_loyalty/static/tests/tours/utils/pos_loyalty_util.js

```diff
@@ -156,11 +156,19 @@ export function createManualGiftCard(code, amount) {
             trigger: `input[id="amount"]`,
             run: `edit ${amount}`,
         },
-        {
-            trigger: `.btn-primary:contains("Add Balance")`,
-            run: "click",
-        },
     ];
+    if (date) {
```

Comment:
Unfortunately the empty string `""` is considered false so it's currently not changing the date in the tour. You can use the following hack, it will check if the type matches as well. An since a string is not a boolean than it's going to go inside the condition.
```suggestion
    if (date !== false) {
```

Reply:
You are absolutely right!